utils/time.go

```diff
@@ -1,23 +1,33 @@
 package utils
 
-import "time"
+// May 18th 2033 in seconds (Jan 24th 1970 if understood as milliseconds)
+const maxSeconds = 2000000000
+// May 18th 2033 in millis (Jan 24th 1970 if understood as microseconds)
+const maxMilliseconds = maxSeconds * 1000
+// May 18th 2033 in microseconds (Jan 24th 1970 if understood as nanoseconds)
+const maxMicroseconds = maxMilliseconds * 1000
 
+// TimeToMillis transforms an integer with arbitrary units into milliseconds, by
+// doing the following assumptions, in this specific order:
+//
+//   * If the number is smaller than the number of milliseconds representing the
+//     date Jan 24th 1970, then the units are assumed to be [seconds]
+//   * If the number is smaller than the number of microseconds representing the
+//     date Jan 24th 1970, then the units are assumed to be [milliseconds]
+//   * If the number is smaller than the number of nanosecondds representing the
+//     date Jan 24th 1970, then the units are assumed to be [microseconds]
+//   * Otherwise, the units are assumed to be [nanoseconds]
+//
+// WARNING: when the current date becomes May 18th 2033, the function below will
```

Comment:
I increased it to 3000000000 as suggested, as the threshold if the date is understood as seconds doesn't vary much (Feb 4th 1970, in this case) and will allow the function to work correctly for 45 years, until 2065. If we change it to 4000000000, it will represent Feb 16th 1970 in seconds, and would last until 2096. Hmmm... Well, we can change it again in 45 years, in case we need it :-P

Reply:
@fridgei , do you know if there was  any business reason to choose 2000000000 in the first place? In case there is, we can just revert the commit  (I did the change in an individual commit).